hack/update-gofmt.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+# Copyright 2014 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# GoFmt apparently is changing @ head...
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${REPO_ROOT}"/hack/util.sh
+
+util::verify_go_version
+
+cd "${REPO_ROOT}"
+
+find_files() {
+  find . -not \( \
+      \( \
+        -wholename './output' \
+        -o -wholename './.git' \
+        -o -wholename './_output' \
```

Comment:
add this folder additionally:
https://github.com/karmada-io/karmada/blob/3674e9f5a85cf8afd7039074d3acb8fd98487f5a/.gitignore#L13

Reply:
why?

Reply:
when will this dir appear?

Reply:
`update-codegen.sh` will create this folder. I am sure what will happen when both this scripts are running at the same time.

Reply:
Besides `_go` shall also be ignored. It's created in `update-codegen.sh`.
This dir is missing in `.gitignore`. you can add it.

Reply:
I see `verify-codegen.sh` will create this dir, not `update-codegen.sh`. But it will delete it afer finish, so I think there is not necessary

Reply:
got it.